pandas/core/arrays/datetimes.py

```diff
@@ -2398,6 +2398,7 @@ def generate_range(start=None, end=None, periods=None, offset=BDay()):
 
     elif end and not offset.is_on_offset(end):
         end = offset.rollback(end)
+        end = Timestamp(str(end).split(' ')[0] + ' 23:59:59')
```

Comment:
I think this fix is too specific. The bug probably lies in not checking the end bounds